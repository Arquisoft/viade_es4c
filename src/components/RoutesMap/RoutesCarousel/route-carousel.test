import React from 'react';
import {cleanup, render} from 'react-testing-library';
import RoutesCarousel from './index';
import { ItemViade, RouteViade } from '../../../viade/Model';

describe.only('RoutesCarousel', () => {
  afterAll(cleanup);

  let item1 = new ItemViade(50,50,1)
    let item2 = new ItemViade(50,50,2)
    let item3 = new ItemViade(50,50,3)
    let items = [item1, item2, item3]

    let prueba = new RouteViade("rutaPrueba", items, "descripcion", [], [])

  const { container } = render(
      <RoutesCarousel route = {prueba.media}/>
  );


  test('renders without crashing', () => {
    expect(container).toBeTruthy();
  });
});