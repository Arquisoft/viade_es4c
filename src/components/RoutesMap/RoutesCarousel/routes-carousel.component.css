.route-img {
    width: 500px;
    height: 500px;
    margin: 0 auto;
}

.carousel-caption {
    background: rgba(188, 218, 211,.8);
}

.carousel-control-prev {
    background: rgba(0,0,0,.2);
}

.carousel-control-next {
    background: rgba(0,0,0,.2);
}