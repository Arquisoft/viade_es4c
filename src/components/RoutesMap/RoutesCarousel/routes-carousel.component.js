import React from "react";
import Carousel from "react-bootstrap/Carousel";
import "./routes-carousel.component.css";


/**
 * Slide with the images of the route
 */
class RoutesCarousel extends React.Component {

	constructor(props) {
		super(props);
		this.data = props.media;
	}

	render() {
		return (
			<Carousel interval={5000}>
				{this.data.map(function (object) {
					return (
						<Carousel.Item key={object.getUrl()}>
							{object.getComponent()}
						</Carousel.Item>);
				})}
			</Carousel>
		);
	}

}

export default RoutesCarousel;

