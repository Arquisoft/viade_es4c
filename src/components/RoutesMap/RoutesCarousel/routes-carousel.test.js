import React from 'react';
import { shallow } from 'enzyme';
import RoutesCarousel from "./routes-carousel.component";

describe('RoutesCarousel', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<RoutesCarousel debug />);
  
    expect(component).toMatchSnapshot();
  });
});
