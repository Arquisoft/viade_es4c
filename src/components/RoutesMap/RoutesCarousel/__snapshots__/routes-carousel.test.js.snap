// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoutesCarousel should render correctly in "debug" mode 1`] = `
<Carousel
  controls={true}
  defaultActiveIndex={0}
  fade={false}
  indicators={true}
  interval={5000}
  keyboard={true}
  nextIcon={
    <span
      aria-hidden="true"
      className="carousel-control-next-icon"
    />
  }
  nextLabel="Next"
  pause="hover"
  prevIcon={
    <span
      aria-hidden="true"
      className="carousel-control-prev-icon"
    />
  }
  prevLabel="Previous"
  slide={true}
  touch={true}
  wrap={true}
>
  <CarouselItem
    key="/img/inrupt.svg"
  >
    <img
      alt="Imagen 1"
      className="d-block route-img"
      src="/img/inrupt.svg"
    />
    <CarouselCaption>
      <h3>
        Imagen 1
      </h3>
    </CarouselCaption>
  </CarouselItem>
  <CarouselItem
    key="/img/logo.svg"
  >
    <img
      alt="Imagen 2"
      className="d-block route-img"
      src="/img/logo.svg"
    />
    <CarouselCaption>
      <h3>
        Imagen 2
      </h3>
    </CarouselCaption>
  </CarouselItem>
  <CarouselItem
    key="/img/Solid.svg"
  >
    <img
      alt="Imagen 3"
      className="d-block route-img"
      src="/img/Solid.svg"
    />
    <CarouselCaption>
      <h3>
        Imagen 3
      </h3>
    </CarouselCaption>
  </CarouselItem>
</Carousel>
`;
